Formula/amp.rb

```diff
@@ -19,7 +19,15 @@ class Amp < Formula
   uses_from_macos "libiconv"
 
   def install
-    ENV.append_to_cflags "-arch #{Hardware::CPU.arch}" if Hardware::CPU.arm?
+    # Using Homebrew's shim breaks building `onig_sys` so force not using
+    # it here, ensuring we add the -arch argument the shim would normally
+    # provide.
+    # https://github.com/Homebrew/homebrew-core/pull/75790
+    if Hardware::CPU.arm?
+      ENV["CC"] = "/usr/bin/clang"
+      ENV["CXX"] = "/usr/bin/clang"
+      ENV.append_to_cflags "-arch #{Hardware::CPU.arch}"
```

Comment:
```suggestion
      ENV["CFLAGS"] = "-arch #{Hardware::CPU.arch}"
```
Now you definitely need this, I think, since `append_to_cflags` only makes it to the compiler via the shims.

Reply:
`append_to_cflags` worked locally weirdly, but I'll change it if you think CI will need that. I don't think CI will even let this run because it doesn't like us modifying the `CC` to a hardcoded variable, but `ENV.cc` resolves to the shim... which is precisely what we're trying to avoid 😅.

Other formulae work around this by using an `inreplace` instead of `ENV` when hiding from our shims but I don't think that's feasible here.